src/cli/perf_pk_kem.cpp

```diff
@@ -0,0 +1,137 @@
+
+#if defined(BOTAN_HAS_PUBLIC_KEY_CRYPTO)
+
+class PerfTest_PK_KEM : public PerfTest {
+   public:
+      virtual std::string algo() const = 0;
+
+      virtual std::vector<std::string> keygen_params(const PerfConfig& config) const {
+         BOTAN_UNUSED(config);
+         return {""};
+      }
+
+      virtual std::string format_name(const std::string& alg, const std::string& param) const {
+         return param.empty() ? alg : Botan::fmt("{}-{}", alg, param);
+      }
+
+      void go(const PerfConfig& config) override {
+         const std::string alg = this->algo();
+
+         const auto params = this->keygen_params(config);
+
+         for(const auto& param : params) {
+            const std::string nm = this->format_name(alg, param);
+            bench_pk_kem(config, nm, alg, param);
+         }
+      }
+
+      void bench_pk_kem(const PerfConfig& config,
+                        const std::string& nm,
+                        const std::string& algo,
+                        const std::string& params,
+                        const std::string& provider = "") {
+         const auto msec = config.runtime();
+         auto& rng = config.rng();
+
+         const std::string kdf = "KDF2(SHA-256)";  // arbitrary choice
+
+         auto keygen_timer = config.make_timer(nm, 1, "keygen");
+
+         auto key = keygen_timer->run([&] { return Botan::create_private_key(algo, rng, params); });
+
+         Botan::PK_KEM_Decryptor dec(*key, rng, kdf, provider);
+         Botan::PK_KEM_Encryptor enc(*key, kdf, provider);
+
+         auto kem_enc_timer = config.make_timer(nm, 1, "KEM encrypt");
+         auto kem_dec_timer = config.make_timer(nm, 1, "KEM decrypt");
+
+         while(kem_enc_timer->under(msec) && kem_dec_timer->under(msec)) {
+            Botan::secure_vector<uint8_t> salt = rng.random_vec(16);
+
+            kem_enc_timer->start();
+            const auto kem_result = enc.encrypt(rng, 64, salt);
+            kem_enc_timer->stop();
+
+            kem_dec_timer->start();
+            Botan::secure_vector<uint8_t> dec_shared_key = dec.decrypt(kem_result.encapsulated_shared_key(), 64, salt);
+            kem_dec_timer->stop();
+
+            if(kem_result.shared_key() != dec_shared_key) {
+               config.error_output() << "KEM mismatch in PK bench\n";
+            }
+         }
+
+         config.record_result(*kem_enc_timer);
+         config.record_result(*kem_dec_timer);
+      }
+};
+
+#endif
+
+#if defined(BOTAN_HAS_KYBER) || defined(BOTAN_HAS_KYBER_90S)
+
+class PerfTest_Kyber final : public PerfTest_PK_KEM {
+   public:
+      std::string algo() const override { return "Kyber"; }
+
+      std::vector<std::string> keygen_params(const PerfConfig& config) const override {
+         BOTAN_UNUSED(config);
+         return {
+            "Kyber-512-r3",
+            "Kyber-512-90s-r3",
+            "Kyber-768-r3",
+            "Kyber-768-90s-r3",
+            "Kyber-1024-r3",
+            "Kyber-1024-90s-r3",
```

Comment:
Admittedly, a bit late to the party. Nevertheless: When building with `--disable-modules=kyber_90s`, this will error with:
```
Kyber-Kyber-512-r3 65559 KEM encrypt/sec; 0.02 ms/op 38070 cycles/op (26176 ops in 399 ms)
Kyber-Kyber-512-r3 52351 KEM decrypt/sec; 0.02 ms/op 47679 cycles/op (26176 ops in 500 ms)
Error: Not implemented requested Kyber mode is not enabled in this build
```

... instead of skipping the Kyber 90s modes. Similarly for other algorithms where specific groups of modes may be disabled at build time.

Could be fixed by (for instance):

```C++
#if defined(BOTAN_HAS_KYBER)
   "Kyber-512-r3",
   "Kyber-768-r3",
   "Kyber-1024-r3",
#endif
#if defined(BOTAN_AS_KYBER_90S)
   "Kyber-512-90s-r3",
   "Kyber-768-90s-r3",
   "Kyber-1024-90s-r3",
#endif
```